extension/core_functions/scalar/debug/sleep.cpp

```diff
@@ -0,0 +1,72 @@
+#include "core_functions/scalar/debug_functions.hpp"
+
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_constant_expression.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include <thread>
+#include <chrono>
+
+namespace duckdb {
+
+static void SleepFunction(DataChunk &input, ExpressionState &state, Vector &result) {
+	UnifiedVectorFormat vdata;
+	input.data[0].ToUnifiedFormat(input.size(), vdata);
+
+	auto milliseconds = UnifiedVectorFormat::GetData<int64_t>(vdata);
+	
+	for (idx_t i = 0; i < input.size(); i++) {
+		auto idx = vdata.sel->get_index(i);
+		if (!vdata.validity.RowIsValid(idx)) {
+			FlatVector::SetNull(result, i, true);
+			continue;
+		}
+		// Sleep for the specified number of milliseconds (clamp negative values to 0)
+		int64_t sleep_ms = milliseconds[idx];
+		if (sleep_ms < 0) {
+			sleep_ms = 0;
+		}
+		std::this_thread::sleep_for(std::chrono::milliseconds(sleep_ms));
+		FlatVector::SetNull(result, i, true);
+	}
+}
+
+static unique_ptr<FunctionData> SleepBind(ClientContext &context, ScalarFunction &bound_function,
+                                          vector<unique_ptr<Expression>> &arguments) {
+	// Validate that we have exactly one argument
+	D_ASSERT(arguments.size() == 1);
+	
+	// Return type is always SQLNULL
+	bound_function.return_type = LogicalType::SQLNULL;
+	
+	return nullptr;
+}
+
+static unique_ptr<Expression> BindSleepFunctionExpression(FunctionBindExpressionInput &input) {
+	if (input.children[0]->IsFoldable()) {
+		Value value = ExpressionExecutor::EvaluateScalar(input.context, *input.children[0]);
```

Comment:
I don't think the constant optimization is necessary. In Postgres, for example, the sleep function just runs once for every invocation:

```sql
\timing on
select pg_sleep(1) from generate_series(0,9);
 pg_sleep 
----------
 
 
 
 
 
 
 
 
 
 
(10 rows)

Time: 10011.742 ms (00:10.012)
```

If you want to only sleep once it's easy to only invoke the function once (`select sleep(...)`).